### PR TITLE
vimPlugins.one-small-step-for-vimkind: init at 2024-11-07

### PR DESCRIPTION
**Pull request by phrmendes**
one-small-step-for-vimkind a.k.a. osv is an adapter for the Neovim lua language. It allows you to debug any lua code running in a Neovim instance.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/vim/plugins/generated.nix [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-{ lib, buildVimPlugin, buildNeovimPlugin, fetchFromGitHub, fetchgit }:
-
-final: prev:
 {
+  lib,
+  buildVimPlugin,
+  buildNeovimPlugin,
+  fetchFromGitHub,
+  fetchgit,
+}: final: prev: {
```
Comment (GaetanLepage) [inline pkgs/applications/editors/vim/plugins/generated.nix:8]: Could you remove this change please ?

Comment (phrmendes) [reply pkgs/applications/editors/vim/plugins/generated.nix]: Done.

File: pkgs/applications/editors/vim/plugins/generated.nix [hunk 2]
```diff
@@ -9930,6 +9930,18 @@ final: prev:
     meta.homepage = "https://github.com/Almo7aya/openingh.nvim/";
   };
 
+  one-small-step-for-vimkind = buildVimPlugin {
+    pname = "one-small-step-for-vimkind";
+    version = "2024-11-07";
```
Comment (khaneliman) [inline pkgs/applications/editors/vim/plugins/generated.nix:9935]:
```suggestion
    version = "2024-10-15";
```
https://github.com/jbyuki/one-small-step-for-vimkind/commit/ad065ad2c814249cfb9e344ce5b2b35d36fbc09f

**Review by khaneliman — APPROVED** [review]
LGTM, thanks!

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354337`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.one-small-step-for-vimkind</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.one-small-step-for-vimkind</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.one-small-step-for-vimkind</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.one-small-step-for-vimkind</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Review by GaetanLepage — APPROVED** [review]

**Pull request merged**